dependencies.yaml

```diff
@@ -57,16 +57,16 @@ dependencies:
           - cython>=0.29,<0.30
           - dlpack>=0.5,<0.6.0a0
           - ninja
-          - &pyarrow_ver pyarrow=9.0.0
+          - pyarrow=10
```

Comment:
Porting my question from the env file: is there a reason that this is only a major version now?

Reply:
To allow floating forward on patch versions since they're ABI compatible.